core/chains/evm/gas/models.go

```diff
@@ -33,11 +33,11 @@ type EvmFeeEstimator interface {
 
 	// L1Oracle returns the L1 gas price oracle only if the chain has one, e.g. OP stack L2s and Arbitrum.
 	L1Oracle() rollups.L1Oracle
-	GetFee(ctx context.Context, calldata []byte, feeLimit uint64, maxFeePrice *assets.Wei, opts ...feetypes.Opt) (fee EvmFee, chainSpecificFeeLimit uint64, err error)
+	GetFee(ctx context.Context, calldata []byte, feeLimit uint64, maxFeePrice *assets.Wei, toAddress *common.Address, opts ...feetypes.Opt) (fee EvmFee, chainSpecificFeeLimit uint64, err error)
```

Comment:
Perhaps using `common.Address` instead of a pointer simplifies the API?

Reply:
I kept this as a pointer so `GetFee` callers that don't care about the gas limit can pass a `nil` like in the case of the ChainWriter.

Reply:
Makes sense, but we also need to add a check [here](https://github.com/smartcontractkit/chainlink/pull/14041/files#diff-77309b300a65b53d3278ff608bf85504b8a3bfc85b5cc9d5085bc36eda2ec0ccR352) to make sure the address is properly set.

Reply:
The `To` field in `ethereum.CallMsg` is actually `*common.Address` so we can pass this directly in whether it is set or not.

Reply:
Yes, but I'm guessing if we do and it's empty it's going to use the 0x0 which might not be the intended behavior?